recipes/lua/all/CMakeLists.txt

```
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
```

```suggestion
cmake_minimum_required(VERSION 3.15)
```

if the CMakeLists is in _our_ repo, let's fix this to 3.15 (the minimum supported by Conan 2)